### PR TITLE
Feature/extend avro

### PR DESCRIPTION
**Pull request by nshumoogum**
Extension to go-ns avro package methods, `marshal` and `unmarshal`.

Better test coverage of avro `marshal` method and it's inherent functions.

Issue https://github.com/ONSdigital/go-ns/issues/18 raised regarding test coverage for avro `unmarshal` method and its inherent functions.

### CODE REVIEW

File: avro/avro.go [hunk 1]
```diff
@@ -106,6 +114,7 @@ func checkFieldType(v reflect.Value, t reflect.Type) error {
 		fieldType := t.Field(i)
 
 		if !isValidType(fieldType.Type.Kind()) {
+			log.Trace("type is:", log.Data{"field_type": fieldType.Type.Kind()})
```
Comment (Matt-Guest) [inline avro/avro.go]: Should we be logging messages in the marshal function?

File: avro/avro_test.go [hunk 2]
```diff
@@ -4,179 +4,342 @@ import (
...
-	message, err := marshalSchema.Marshal(data)
-	if err != nil {
-		return nil, err
+// TODO -- -- Test unmarshalStringSlice function
+
+// TODO -- -- Test unmarshalStructSlice function
+
+// TODO -- -- -- Test populateNestedArrayItem function
+
+// TODO -- -- -- -- Test setNestedStructs function
```
Comment (Matt-Guest) [inline avro/avro_test.go]: There a section of TODO comments here. looks like you done the tests for them. Can they be removed?

File: avro/test_data.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    ]
+}
+`
+
+type testData struct {
+	Manager         string `avro:"manager"`
+	TeamName        string `avro:"team_name"`
+	Owner           string `avro:"ownerOfTeam"`
+	Sport           string `avro:"kind-of-sport"`
+	Uri             string `avro:"uri"`
```
Comment (Matt-Guest) [inline avro/test_data.go]: should this be URI?

File: avro/avro_test.go (hunk 2 above)
Comment (nshumoogum) [inline avro/avro_test.go]: All these need to be written, they're just place holders for now.

File: avro/test_data.go (hunk 3 above)
Comment (nshumoogum) [inline avro/test_data.go]: yes 👍

**Pull request merged**